src/pages/PrintOnlyScreen.tsx

```diff
@@ -62,9 +62,8 @@ const PrintOnlyScreen = ({
     !isPrinted
 
   const printBallot = useCallback(async () => {
-    // get the tracker first
-    const tracker = await encryptAndGetTracker(votes)
-    setTrackerString(tracker)
+    const ballotTrackingCode = await getBallotTrackingCode(votes)
+    setTrackerString(ballotTrackingCode)
```

Comment:
@danwallach flagged an interesting issue above that made me wonder about `trackerString`. I'm guessing that if `markVoterCardPrinted` were synchronous that `trackerString` would always be empty when printing. As-is, I think it constitutes a race condition since React rendering after mutating state is also asynchronous.

Reply:
Ahah, good catch! What would be the right way to wait until the re-rendering?

Reply:
I decided to model this as a state machine like so:

```
Wait for Card
- card inserted with votes -> Encrypt Ballot
- card inserted without votes -> Empty Card

Encrypt Ballot
- ballot encrypt succeeded -> Mark Card Used
- ballot encrypt failed -> ???

Mark Card Used
- erase card succeeded -> Print
- erase card failed -> ???

Print
- print started -> Update Tally
- print failed to start -> ???

Update Tally
- update tally succeeded -> Printing
- update tally failed -> ???

Printing
- print succeeded -> Verify Printed Ballot
- print failed -> ???

Verify Printed Ballot
- card removed -> Wait for Card

Empty Card
- card removed -> Wait for Card
```

Each state is named e.g. `Empty Card` and represents a particular React render. The actions that can happen in that state are enumerated below the name along with the state they should transition to. Notice that there are several error handling states that are missing.

> Ahah, good catch! What would be the right way to wait until the re-rendering?

I think we need to set the tracker code state and then, in another render, trigger the print. While I don't think we _have_ to make the states more explicit in this PR, I do think it might help.